internal/parser.go

```diff
@@ -1553,7 +1553,7 @@ func (p *parser) inBodyEndTagFormatting(tagAtom a.Atom, tagName string) {
 	// refactor this code to be more idiomatic.
 
 	// Steps 1-2
-	if current := p.oe.top(); current.Data == tagName && p.afe.index(current) == -1 {
+	if current := p.oe.top(); current != nil && current.Data == tagName && p.afe.index(current) == -1 {
```

Comment:
this is the fix for the "weird control characters" test. i feel like we should error instead of returning a nil node here though?